src/Autocomplete.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  query: string;
+  setQuery: (query: string) => void;
+  onSelected: (person: Person) => void;
+  onQueryChange: (query: string) => void;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({ people, query, setQuery, onSelected, onQueryChange }) => {
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  useEffect(() => {
+    setIsDropdownActive(query === '' || people.length > 0);
+  }, [query, people]);
+
+  return (
+    <div className={`dropdown ${isDropdownActive ? 'is-active' : ''}`}>
+          <div className="dropdown-trigger">
+            <input
+              type="text"
+              placeholder="Enter a part of the name"
+              className="input"
+              value={query}
+              onChange={(e) => {
+                setQuery(e.target.value);
+                onQueryChange(e.target.value);
```

Comment:
Make sure that the filter won't be called if the user enters spaces only. This is a requirement from the checklist: 'make sure that filter won't be called if user entered spaces only'.